compiler/src/openxla/compiler/nvgpu/Conversion/ConvertHLOToCUDNN.cpp

```diff
@@ -224,18 +226,21 @@ static LogicalResult convertConv(stablehlo::ConvolutionOp op,
 
   uint32_t spatialDimCount =
       op.getDimensionNumbers().getInputSpatialDimensions().size();
-  auto getAttrOr = [spatialDimCount](std::optional<DenseIntElementsAttr> attr,
-                                     int64_t value) {
-    if (!attr) return SmallVector<int64_t>(spatialDimCount, value);
-    SmallVector<int64_t> values;
-    llvm::transform(attr->getValues<APInt>(), std::back_inserter(values),
-                    [](APInt stride) { return stride.getSExtValue(); });
-    return values;
+  auto getAttrOr = [](std::optional<DenseIntElementsAttr> attr, int64_t size,
+                      int64_t value) -> SmallVector<int64_t> {
+    if (!attr) return SmallVector<int64_t>(size, value);
+    return llvm::to_vector(llvm::map_range(
+        attr->getValues<APInt>(), [](APInt it) { return it.getSExtValue(); }));
   };
-  SmallVector<int64_t> spatialStride = getAttrOr(op.getWindowStrides(), 1);
-  SmallVector<int64_t> prePadding = getAttrOr(op.getPadding(), 0);
-  SmallVector<int64_t> postPadding(spatialDimCount, 0);
-  SmallVector<int64_t> dilation = getAttrOr(op.getRhsDilation(), 1);
+  SmallVector<int64_t> spatialStride =
+      getAttrOr(op.getWindowStrides(), spatialDimCount, 1);
+  SmallVector<int64_t> padding =
+      getAttrOr(op.getPadding(), 2 * spatialDimCount, 0);
+  assert(padding.size() == 2 * spatialDimCount);
+  ArrayRef<int64_t> prePadding = ArrayRef(padding).take_front(spatialDimCount);
```

Comment:
Are you sure this is correct? E.g. for `tensor<2x2xi32>` isn't it `padding[0] = {pre_padding_dim0, post_padding_dim0}` and padding[1] = `{ pre_..0, post_...1}` ?

Reply:
Thanks for checking